<launch>

    <!-- Enable custom logger configuration-->
    <env name="ROS_PYTHON_LOG_CONFIG_FILE" value="$(find mapc_rhbp_ettlinger)/launch/python_logging.conf"/>

    <!--Infrastructure Nodes-->
    <include file="$(find mapc_rhbp_ettlinger)/launch/graphhopper.launch">
        <arg name="initial_port" value="8989" />
    </include>

    <!--Planner Nodes-->
    <include file="$(find mapc_rhbp_ettlinger)/launch/planner.launch" pass_all_args="true">
    </include>

    <include file="$(find mapc_rhbp_ettlinger)/launch/survailer.launch" pass_all_args="true">
    </include>

    <include file="$(find mapc_rhbp_ettlinger)/launch/debug_agent.launch" pass_all_args="true">
    </include>

    <!-- Drones -->
        <!-- Bridges -->
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA1" />
        <arg name="only_agent_specific" value="False" />
    </include>

    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA2" />
    </include>

    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA3" />
    </include>

    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA4" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA5" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA6" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA7" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA8" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA9" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA10" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA11" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA12" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA13" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA14" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA15" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA16" />
    </include>
    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA17" />
    </include>

    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA1" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA2" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA3" />
        <arg name="bid_for_auctions" value="True" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA4" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA5" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA6" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA7" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA8" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA9" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA10" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA11" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA12" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA13" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA14" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA15" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA16" />
    </include>
    <include file="$(find mapc_rhbp_ettlinger)/launch/rhbp_agent.launch" pass_all_args="true">
        <arg name="agent_name" value="agentA17" />
    </include>

    <param name="situationBias" type="double" value="1.0" />
    <param name="predecessorBias" type="double" value="1.0" />
    <param name="successorBias" type="double" value="1.0" />
    <param name="conflictorBias" type="double" value="1.0" />
    <param name="goalBias" type="double" value="1.0" />
    <param name="planBias" type="double" value="1.0" />
    <param name="activationThreshold" type="double" value="7.0" />
    <param name="createLogFiles" type="bool" value="false"/>
    <param name="activation_algorithm" type="string" value="default"/>
    <param name="max_parallel_behaviours" type="int" value="1"/> <!--Comment or negative for unlimited-->

    <param name="RhbpAgent.MAX_DECISION_MAKING_TIME" type="double" value="14"/>
    <param name="RhbpAgent.BUILD_WELL_ENABLED" type="bool" value="True"/>


    <param name="ShouldBidForAssemblyDecision.WEIGHT_LOAD" type="double" value="30"/>
    <param name="ShouldBidForAssemblyDecision.WEIGHT_INGREDIENT_LOAD" type="double" value="100"/>
    <param name="ShouldBidForAssemblyDecision.WEIGHT_STEPS" type="double" value="-3"/>
    <param name="ShouldBidForAssemblyDecision.ACTIVATION_THRESHOLD" type="double" value="-15"/>

    <param name="ChooseStorageForHoardingDecision.WEIGHT_STEPS"  value="-1" type="double"/>
    <param name="ChooseStorageForHoardingDecision.WEIGHT_ITEMS_ALREADY_THERE"  value="-1" type="double"/>

    <param name="ChooseItemToGatherMechanism.WEIGHT_STEPS"  value="1" type="double"/>
    <param name="ChooseItemToGatherMechanism.WEIGHT_ASSEMBLY_ROLE_MATCH"  value="2.5" type="double"/>
    <param name="ChooseItemToGatherMechanism.WEIGHT_ASSEMBLY_ROLE_MATCH_COUNT"  value="0.5" type="double"/>
    <param name="ChooseItemToGatherMechanism.WEIGHT_PRIORITY"  value="200" type="double"/>
    <param name="ChooseItemToGatherMechanism.FINISHED_PRODUCT_PRIORITY_TO_INGREDIENT_CONVERSION"  value="3" type="double"/>


    <param name="ChooseBestAvailableJobDecision.PERCNTILE_TO_TRY_JOB"  value="0.7" type="double"/>
    <param name="ChooseBestAvailableJobDecision.BID_PERCENTILE"  value="50" type="int"/>
    <param name="ChooseBestAvailableJobDecision.TIME_LEFT_WEIGHT_START"  value="30" type="int"/>
    <param name="ChooseBestAvailableJobDecision.ACTIVATION_THRESHOLD"  value="-50" type="double"/>
    <param name="ChooseBestAvailableJobDecision.IMPORTANT_JOB_PERCENTILE"  value="0.95" type="double"/>
    <param name="ChooseBestAvailableJobDecision.ACTIVATION_TO_DESIRED_PRODUCT_CONVERSION"  value="0.2" type="double"/>
    <param name="ChooseBestAvailableJobDecision.WEIGHT_PERCENTILE"  value="10.0" type="double"/>
    <param name="ChooseBestAvailableJobDecision.WEIGHT_TIME_PASSED"  value="-0.3" type="double"/>
    <param name="ChooseBestAvailableJobDecision.WEIGHT_TIME_OVER"  value="-0.4" type="double"/>

    <param name="ChooseBestJobAgentCombinationDecision.MIN_STEP_BUFFER"  value="5" type="double"/>
    <param name="ChooseBestJobAgentCombinationDecision.WEIGHT_LOAD"  value="8" type="double"/>
    <param name="ChooseBestJobAgentCombinationDecision.WEIGHT_INGREDIENT_LOAD"  value="8" type="double"/>
    <param name="ChooseBestJobAgentCombinationDecision.WEIGHT_NO_STORAGE_NEEDED"  value="20" type="double"/>
    <param name="ChooseBestJobAgentCombinationDecision.WEIGHT_STEPS"  value="-1" type="double"/>
    <param name="ChooseBestJobAgentCombinationDecision.ACTIVATION_THRESHOLD"  value="-1000" type="double"/>

    <param name="ContractNetManager.DEADLINE_BIDS"  value="2" type="double"/>
    <param name="ContractNetManager.DEADLINE_ACKNOWLEDGEMENT"  value="2" type="double"/>

    <param name="MainAssembleAgentDecision.WEIGHT_FINISHED_PRODUCT_FACTOR"  value="-1" type="double"/>
    <param name="MainAssembleAgentDecision.WEIGHT_BID_SKILL"  value="-100" type="double"/>
    <param name="MainAssembleAgentDecision.WEIGHT_BID_SPEED"  value="100" type="double"/>



    <param name="BestAgentAssemblyCombinationDecision.WEIGHT_NUMBER_OF_AGENTS" type="double" value="-0.2"/>
    <param name="BestAgentAssemblyCombinationDecision.MAX_NR_OF_AGENTS_TO_CONSIDER" type="int" value="17"/>
    <param name="BestAgentAssemblyCombinationDecision.MAX_PRIORITY_NOT_NEEDED_ITEMS" type="double" value="0.30"/>
    <param name="BestAgentAssemblyCombinationDecision.MAX_COUNT_NOT_NEEDED_ITEMS" type="int" value="7"/>

    <param name="BestAgentAssemblyCombinationDecision.MAX_AGENTS" type="int" value="7"/>
    <param name="BestAgentAssemblyCombinationDecision.MIN_AGENTS" type="int" value="3"/>
    <param name="BestAgentAssemblyCombinationDecision.MAX_STEPS" type="int" value="20"/>
</launch>